flexget/plugins/input/telegram_input.py

```diff
@@ -0,0 +1,228 @@
+            },
+        },
+        'required': ['token'],
+        'additonalProperties': False,
+    }
+
+    @plugin.internet(logger)
+    def on_task_input(self, task, config):
+        #Load The Configs
+        token = config['token']
+        clean_update = config['clean_update']
+        entry_config = config.get('entry')
+        whitelist = config.get('whitelist')
+        types = config.get('types')
+
+        #Defaults
+        if not types:
+            types = ['private', 'group']
+
+        #Get Last Checked ID
+        update_id = task.simple_persistence.get(token+'_update_id')
+
+        #Set Last Checked ID as Offset
+        offset = ""
+        if update_id:
+            update_id += 1
+            offset = "?offset="+str(update_id)
+        
+        #The Target URL
+        url = _TELEGRAM_API_URL + "bot" + token + "/getUpdates" + offset  
+
+        #Get Telegram Updates
+        try:
+            response = task.requests.get(
+                url, timeout=60, raise_status=False
+            ).json()
+        except:
+            logger.error("Error getting telegram update")
+            return
+
+        #We have a error
+        if not response['ok']:
+            logger.error("Telegram updater returned error {}: {}", response['error_code'], response['description'] )
+            return
+
+        #Get All New Messages
+        messages = response['result']
+
+        entries = []
+        for message in messages:
+            #This is the ID
+            update_id = message['update_id']
+
+            #If desired update the last ID for the Bot
+            if clean_update:
+                logger.debug("Last Update set to {}",update_id)
+                task.simple_persistence[token+'_update_id'] = update_id
+            
+            #We Don't care if it's not a message or no text
+            if 'message' not in message:
+                continue
+
+            if 'text' not in message['message']:
+                continue
+
+            if 'chat' not in message['message']:
+                continue
+
+            if 'type' not in message['message']['chat']:
+                continue
+
+            logger.debug("Income message: {}", message)
+
+            #Create Base Entry
+            text = message['message']['text']
+            entry = Entry()
+            entry['title'] =  text
+
+            #We need a url, so we add a dummy
+            entry['url'] = "http://localhost?update_id="+str(update_id)
+
+            #Store the message if we need to use it in other plugins
+            entry['telegram_message'] =  message['message']
+            
+            #Check Types
+            if types:
+                if message['message']['chat']['type'] not in types:
+                    logger.debug("Ignoring message because of invalid type {}",message)
+                    continue
+
+            #Check From
+            append = True
+            if whitelist:
```

Comment:
you can set the default for whitelist to an empty list and skip checking for truthy, just iterate over it

Reply:
I did the default, but I have to keep the if! Because if no whitelist is informed I accept anything. If the if is not checking the existence of the "whitelist" the for will jump to the else and reject the message, and it should accept it